pkg/cmd/pipeline/translate.go

```diff
@@ -0,0 +1,226 @@
+// limitations under the License.
+
+package pipeline
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"math"
+	"os"
+	"strings"
+
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	apiv1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	nameField       = "name"
+	statusField     = "status"
+	outputField     = "output"
+	repoField       = "repository"
+	branchField     = "branch"
+	filenameField   = "filename"
+	yamlField       = "yaml"
+	iconField       = "icon"
+	actionLockField = "actionLock"
+	actionNameField = "actionName"
+
+	actionDefaultName = "cli"
+
+	// ProgressingStatus indicates that an app is being deployed
+	ProgressingStatus = "progressing"
+	// DeployedStatus indicates that an app is deployed
+	DeployedStatus = "deployed"
+	// ErrorStatus indicates that an app has errors
+	ErrorStatus = "error"
+	// DestroyingStatus indicates that an app is being destroyed
+	DestroyingStatus = "destroying"
+
+	// maxLogOutput is the maximum size that we allow to allocate for logs.
+	// Specifically 800kb. The limit on configmaps is 1Mb but we want to leave some
+	// room for the other data stored in there.
+	// Note that the is the limit after encoding the logs to base64 which is how
+	// the logs are stored in the configmap.
+	maxLogOutput = 800 << (10 * 1)
+)
+
+// maxLogOutputRaw is the maximum size we allow to allocate for logs before
+// being encoded to base64
+// See: https://stackoverflow.com/a/4715480/1100238
+var maxLogOutputRaw = int(math.Floor(float64(maxLogOutput)*3) / 4)
+
+//CfgData represents the data to be include in a configmap
+type CfgData struct {
+	Name       string
+	Namespace  string
+	Status     string
+	Output     string
+	Repository string
+	Branch     string
+	Filename   string
+	Manifest   []byte
+	Icon       string
+}
+
+// TranslateConfigMap translates the app into a configMap
+func TranslateConfigMap(ctx context.Context, data *CfgData, c kubernetes.Interface) (*apiv1.ConfigMap, error) {
+	cmap, err := configmaps.Get(ctx, TranslatePipelineName(data.Name), data.Namespace, c)
+	if err != nil {
+		if !oktetoErrors.IsNotFound(err) {
+			return nil, err
+		}
+		cmap = translatedConfigMapSandBox(data)
+		err := configmaps.Create(ctx, cmap, cmap.Namespace, c)
+		if err != nil {
+			if k8sErrors.IsAlreadyExists(err) {
+				return nil, fmt.Errorf("A new deploy operation has started")
+			}
+			return nil, err
+		}
+	}
+
+	if err := updateCmap(cmap, data); err != nil {
+		return nil, err
+	}
+	if err := configmaps.Deploy(ctx, cmap, cmap.Namespace, c); err != nil {
+		if k8sErrors.IsConflict(err) {
+			return nil, fmt.Errorf("Another pipeline operation is being executed")
```

Comment:
I think we need consistency in errors. Here we mention "Another pipeline..." but in other places we mention "A new deploy..." or "...deploy operation already running"

I think we should use `pipeline` or `deploy operation` but using both could be confusing